driver/src/main/java/org/neo4j/driver/internal/net/SocketClient.java

```diff
@@ -133,11 +135,7 @@ public void start()
         }
         catch ( IOException e )
         {
-            throw new ClientException( "Unable to process request: " + e.getMessage(), e );
-        }
-        catch ( GeneralSecurityException e )
-        {
-            throw new ClientException( "Unable to establish ssl connection with server: " + e.getMessage(), e );
+            throw new ServiceUnavailableException( "Unable to process request: " + e.getMessage(), e );
```

Comment:
Could a test be added to assert that router is removed when this exception is thrown?